tests/baselines/reference/esDecorators-classExpression-classSuper.2.errors.txt

```diff
@@ -0,0 +1,32 @@
+esDecorators-classExpression-classSuper.2.ts(7,15): error TS2850: Class field 'name' defined by the parent class is not accessible in the child class via super.
+
+
+==== esDecorators-classExpression-classSuper.2.ts (1 errors) ====
+    declare var dec: any;
+    
+    // class expression in extends should not get an assigned name
+    (@dec
+    class C1 extends class { } {
+        static {
+            super.name;
+                  ~~~~
+!!! error TS2850: Class field 'name' defined by the parent class is not accessible in the child class via super.

```

Comment:
I don’t think this looks right. If we do

```ts
class C1 extends class Expr {} {
    static {
        console.log(super.name);
    }
}
```

this successfully logs `"Expr"`. Another example:

```ts
class C1 extends Number {
    static {
        console.log(super.EPSILON);
    }
}
```

It seems like “is a property declaration in an interface” does not imply “is implemented as a class field” in practice (particularly for built-in type constructors, but I also worry about DefinitelyTyped and other hand-written typings. I think we need to take the more conservative approach of looking for an actual class field declaration.

Reply:
I can exclude the static initialization block for now.

Reply:
It’s not specific to the static case; this executes fine as well:

```ts
class C1 extends Array {
    foo() {
        console.log(super.length);
    }
}

new C1().foo()
```

A bad idea? Probably. But all these built-ins are not classes with class fields; they’re more like old-school constructor functions. So I think we should never issue this error unless we literally see class field syntax.

Reply:
> It’s not specific to the static case; this executes fine as well:
> 
> ```ts
> class C1 extends Array {
>     foo() {
>         console.log(super.length);
>     }
> }
> 
> new C1().foo()
> ```
> 
> A bad idea? Probably. But all these built-ins are not classes with class fields; they’re more like old-school constructor functions. So I think we should never issue this error unless we literally see class field syntax.

Oh... can we trust the class field in a declaration file?

Reply:
I changed it to check its declaration, I think it covers most false positives.

Reply:
Yeah, I think a class field in a declaration file is fair game. I’ll take a look at the changes.

Reply:
I think this looks good; can you add my examples to the tests?